relays/ethereum/src/ethereum_sync_loop.rs

```diff
@@ -80,41 +82,38 @@ struct EthereumHeadersSource {
 
 type EthereumFutureOutput<T> = OwnedSourceFutureOutput<EthereumHeadersSource, EthereumHeadersSyncPipeline, T>;
 
+#[async_trait]
 impl SourceClient<EthereumHeadersSyncPipeline> for EthereumHeadersSource {
 	type Error = ethereum_client::Error;
-	type BestBlockNumberFuture = Pin<Box<dyn Future<Output = EthereumFutureOutput<u64>>>>;
-	type HeaderByHashFuture = Pin<Box<dyn Future<Output = EthereumFutureOutput<Header>>>>;
-	type HeaderByNumberFuture = Pin<Box<dyn Future<Output = EthereumFutureOutput<Header>>>>;
 	type HeaderExtraFuture = Pin<Box<dyn Future<Output = EthereumFutureOutput<(EthereumHeaderId, Vec<Receipt>)>>>>;
```

Comment:
Why is this left out?

Reply:
I think it is related to the question in the PR description - i.e. "Changing those resulted in a firehose of ownership errors." :) I'll take a look

Reply:
:facepalm: Sorry, I'm guilty of not reading the description and jumping straight to files :)